timemachine/fe/mcgregor.py

```diff
@@ -128,6 +128,30 @@ def __init__(self, n_vertices, edges):
             for edge_idx in edges:
                 self.ve_matrix[vertex_idx][edge_idx] = True
 
+    def _verify_partially_connected(self, source, node_states, n_mapped_nodes):
+        """A fast BFS node generator, adapted from _plain_bfs in networkx."""
+        unseen = [1] * self.n_vertices
+        unseen[source] = 0
+        nextlevel = [source]
+        mapped_count = 1
+
+        if mapped_count == n_mapped_nodes:
+            return True
+
+        while nextlevel:
```

Comment:
nit: Lots of truthy checks here, would rather be explicit but don't think there is much risk here